compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -730,21 +756,206 @@ fn function_def_return_type(
+                let rule = "a pattern";
+                InterpreterError::FailedToParseMacro {
+                    error,
+                    tokens: parameter_name_tokens,
+                    rule,
+                    file: location.file,
+                }
+            })?;
+
+        let hir_pattern = interpreter.elaborate_item(interpreter.current_function, |elaborator| {
+            elaborator.elaborate_pattern_and_store_ids(
+                parameter_pattern,
+                parameter_type.clone(),
+                DefinitionKind::Local(None),
+                &mut parameter_idents,
+                None,
+            )
+        });
+
+        parameters.push((hir_pattern, parameter_type.clone(), Visibility::Private));
+        parameter_types.push(parameter_type);
+    }
+
+    let func_meta = interpreter.elaborator.interner.function_meta(&func_id);
+    let function_type = Type::Function(
+        parameter_types,
+        Box::new(func_meta.return_type().clone()),
+        Box::new(Type::Unit),
+    );
+
+    // TODO: generics. In Elaborator::define_function_meta there's this code:
+    //
+    //     if !generics.is_empty() {
+    //         typ = Type::Forall(generics, Box::new(typ));
+    //     }
+
+    interpreter.elaborator.interner.push_definition_type(func_meta.name.id, function_type.clone());
+
+    {
+        let func_meta = interpreter.elaborator.interner.function_meta_mut(&func_id);
+        func_meta.parameters = parameters.into();
+        func_meta.parameter_idents = parameter_idents;
+        func_meta.typ = function_type;
+    }
+
+    Ok(Value::Unit)
+}
+
+// fn set_return_type(self, return_type: Type)
+fn function_def_set_return_type(
+    interpreter: &mut Interpreter,
+    arguments: Vec<(Value, Location)>,
+    location: Location,
+) -> IResult<Value> {
+    let (
+        (self_argument, self_argument_location),
+        (return_type_argument, return_type_argument_location),
+    ) = check_two_arguments(arguments, location)?;
+    let return_type = get_type(return_type_argument, return_type_argument_location)?;
+
+    let func_id = get_function_def(self_argument, self_argument_location)?;
+    let func_meta = interpreter.elaborator.interner.function_meta(&func_id);
+    match func_meta.function_body {
+        FunctionBody::Unresolved(_, _, _) => (),
+        FunctionBody::Resolving | FunctionBody::Resolved => {
+            return Err(InterpreterError::CannotMutateFunction { location })
+        }
+    }
+
+    let parameter_types = func_meta.parameters.iter().map(|(_, typ, _)| typ.clone()).collect();
+
+    let func_meta = interpreter.elaborator.interner.function_meta(&func_id);
+    let function_type =
+        Type::Function(parameter_types, Box::new(return_type.clone()), Box::new(Type::Unit));
+
+    // TODO: generics. In Elaborator::define_function_meta there's this code:
+    //
+    //     if !generics.is_empty() {
+    //         typ = Type::Forall(generics, Box::new(typ));
+    //     }
```

Comment:
Probably easier to edit the existing function type

Reply:
I tried to edit the existing function but for some reason it didn't work, I'm not sure why.

Reply:
Oh, maybe it's because I also removed the `push_definition_type` call, but that should still be done. I'll try it again...

Reply:
It was that! It's working now.